sdv/tabular/base.py

```diff
@@ -398,40 +432,36 @@ def sample(self, num_rows=None, max_retries=100, max_rows_multiplier=10,
             if not isinstance(group, tuple):
                 group = [group]
 
-            condition_index = dataframe['__condition_idx__'].iloc[0]
-            if transformed_conditions.empty:
-                transformed_condition = None
-            else:
-                transformed_condition = transformed_conditions.loc[condition_index].to_dict()
-
+            condition_indices = dataframe['__condition_idx__']
             condition = dict(zip(condition_columns, group))
-            sampled_rows = self._sample_batch(
-                len(dataframe),
-                max_retries,
-                max_rows_multiplier,
-                condition,
-                transformed_condition,
-                float_rtol
-            )
-
-            if len(sampled_rows) < len(dataframe):
-                # Didn't get enough rows.
-                if len(sampled_rows) == 0:
-                    error = 'No valid rows could be generated with the given conditions.'
-                    raise ValueError(error)
-
-                elif not graceful_reject_sampling:
-                    error = f'Could not get enough valid rows within {max_retries} trials.'
-                    raise ValueError(error)
-
-                else:
-                    warn(f'Only {len(sampled_rows)} rows could '
-                         f'be sampled within {max_retries} trials.')
-
-            if len(sampled_rows) > 0:
-                sampled_rows['__condition_idx__'] = \
-                    dataframe['__condition_idx__'].values[:len(sampled_rows)]
+            if transformed_conditions.empty:
+                sampled_rows = self._conditionally_sample_rows(
+                    dataframe,
+                    max_retries,
+                    max_rows_multiplier,
+                    condition,
+                    None,
+                    float_rtol,
+                    graceful_reject_sampling
+                )
                 all_sampled_rows.append(sampled_rows)
+            else:
+                transformed_conditions_in_group = transformed_conditions.loc[condition_indices]
```

Comment:
I wonder if this entire `else` block could be moved into its own method? I'm not completely sure about whether it would make this method more readable or the opposite, so I'm just thinking out loud. Your call @amontanez24

Reply:
I'm torn as well. Moving it into another method would mean passing `all_sampled_rows` to that other method and appending there which I don't love because it would not be obvious within `sample` what is happening to that variable. The only real difference between this block and the other is grabbing the transformed groups and looping through them. I'm inclined to leave it

Reply:
I agree, let's leave it as it is right now.